towhee/dag/dataframe_repr.py

```diff
@@ -12,59 +12,82 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# builder = GraphBuilder()
+# data1 = data0.map(op1)
+# data2 = data1.batch(op2, bs=64)
 
 from collections import OrderedDict
-from towhee.dataframe.dataframe import DFIterator
+from types import FunctionType
+from typing import NamedTuple
+
+from towhee.base_repr import BaseRepr
 from towhee.dag.variable_repr import VariableRepr
 
 
-class DFRepr:
-    """
-    A collection of VariableRepr
+class DataframeRepr(BaseRepr):
+    """`DataframeRepr` represents a single dataframe within a graph. A single dataframe
+    is composed of multiple individual variables, each of which is required in the next
+    operator stage within the graph.
+
+    Args:
+        name:
+            The representation name.
+        vars:
+            A list of variables that can be accessed from within the dataframe.
     """
-    def __init__(self, name = None, vars = None):
-        """
-        Args:
-            name: name of the DFRepr
-            vars: a list of VariableRepr
-        """
-        self.name = name
-        self.columns =  OrderedDict((var.name, var) for var in vars)
-        self._iter = DFIterator(df = None)
 
-    def __getattr__(self, key: str):
-        """
-        Access a VariableRepr as DFRepr's attribute
+    def __init__(self, name: str):
+        super().__init__(name)
+        # `_columns` must be filled in after this representation is instantiated
+        self._columns = OrderedDict()
+
+    def __getitem__(self, key) -> VariableRepr:
+        """Access a variable representation via dictionary-like indexing.
+
         Args:
-            key: the name of the VariableRepr
+            key:
+                Name of the variable representation to access.
+
+        Returns:
+            The variable corresponding to the specified key, or None if an invalid key
+            was provided.
         """
-        return self.columns[key]
+        return self._columns.get(key)
 
     def __setitem__(self, key: str, value: VariableRepr):
-        """
-        Set a VariableRepr
+        """Sets a single variable representation within the dataframe.
+
         Args:
-            key: the name of the VariableRepr
-            value: the object of the VariableRepr
-        """
-        self.columns[key] = value
-    
-    def __getitem__(self, key):
+            key:
+                Variable name.
+            value:
+                A pre-instantiated `VariableRepr` instance.
         """
-        Get a VariableRepr by its name
+        self._columns[key] = value
+
+    def from_input_annotations(self, func: FunctionType):
```

Comment:
In annotation, we should use "typing.Callable", not "types.FunctionType"

Reply:
Fixed.